xblock/internal.py

```diff
@@ -58,8 +63,8 @@ class Nameable(object):
     :class:`.NamedAttributesMetaclass`, will be assigned a `__name__`
     attribute based on what class attribute they are bound to.
     """
-    __slots__ = ('__name__')
-
+    if six.PY2:
+        __slots__ = ('__name__',)
     __name__ = None
```

Comment:
@cpennington  This breaks under python 3 because slots only apply to instance level attributes, and it doesn't like having a class attribute of the same name.  Python 2 seems okay with it, so I left it conditionally.  Note also that the __slots__ was originally mistakenly defined as a string, rather than as a tuple of strings.  The docs are ambivalent about whether that works.